internal/server/models/credentialrequest.go

```diff
@@ -0,0 +1,62 @@
+package models
+
+import (
+	"time"
+
+	"github.com/infrahq/infra/api"
+	"github.com/infrahq/infra/uid"
+)
+
+// CredentialRequest represents a request from a user to get login credentials to a destination.
+type CredentialRequest struct {
```

Comment:
I wonder if we should call this one `DestinationCredential` and the original one `InfraCredential` or `LocalCredential`.